src/module/mod.rs

```diff
@@ -372,12 +374,12 @@ impl Module {
             custom_types: CustomTypesCollection::new(),
             modules: BTreeMap::new(),
             variables: BTreeMap::new(),
-            all_variables: BTreeMap::new(),
-            functions: BTreeMap::new(),
-            all_functions: BTreeMap::new(),
-            dynamic_functions: BTreeSet::new(),
-            type_iterators: BTreeMap::new(),
-            all_type_iterators: BTreeMap::new(),
+            all_variables: StraightHashMap::default(),
```

Comment:
We may want to set the `capacity` in these constructors, since some maps typically have few functions, so it is prudent to reduce the hash table's size.

Reply:
the default `capacity` of HashMap is 0, it usually performs better in scenarios with small amounts of data.